include/seqan3/search/dream_index/technical_binning_directory.hpp

```diff
@@ -0,0 +1,66 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \brief Provides seqan3::technical_binning_directory.
+ */
+
+#pragma once
+
+#include <seqan3/search/dream_index/interleaved_bloom_filter.hpp>
+
+namespace seqan3
+{
+
+struct ibf_config
+{
+    bin_count number_of_bins;
+    bin_size size_of_bin;
+    hash_function_count number_of_hash_functions;
+};
+
+template <typename /*std::ranges::view*/ hash_view_t, data_layout data_layout_mode_ = data_layout::uncompressed>
+class technical_binning_directory : public interleaved_bloom_filter<data_layout_mode_>
+{
+private:
+    using base_t = interleaved_bloom_filter<data_layout_mode_>;
+    hash_view_t hash_view;
+
+public:
+    technical_binning_directory() = default; //!< Defaulted.
+    technical_binning_directory(technical_binning_directory const &) = default; //!< Defaulted.
+    technical_binning_directory & operator=(technical_binning_directory const &) = default; //!< Defaulted.
+    technical_binning_directory(technical_binning_directory &&) = default; //!< Defaulted.
+    technical_binning_directory & operator=(technical_binning_directory &&) = default; //!< Defaulted.
+    ~technical_binning_directory() = default; //!< Defaulted.
+
+    template <typename rng_t>
+    technical_binning_directory(std::vector<rng_t> const & technical_bins,
+                                hash_view_t && hash_view,
```

Comment:
```suggestion
                                hash_view_t hash_view,
```

Because `hash_view_t` is concrete type here.